packages/editor/src/lib/app/managers/HistoryManager.ts

```diff
@@ -176,13 +179,19 @@ export class HistoryManager<CTX extends { emit: (name: string) => void }> {
 					redos = redos.push(mark)
 				}
 				if (mark.id === toMark) {
-					this.ctx.emit('change-history')
+					this.ctx.emit(
+						'change-history',
+						pushToRedoStack ? { reason: 'undo' } : { reason: 'bail', markId: toMark }
```

Comment:
What does `bail` mean?

Reply:
Bail is when we undo back to the most recent mark without adding anything to the redo stack. Used most often when pressing `escape` during an interaction.